build.rs

```diff
@@ -13,4 +13,19 @@ fn main() {
     {
         println!("cargo::rustc-cfg=feature=\"include-ui\"");
     }
+    // Create a new, empty icon collection
+    let mut icon_dir = ico::IconDir::new(ico::ResourceType::Icon);
+    // Read a PNG file from disk and add it to the collection
+    let file = std::fs::File::open("assets/icon.png").expect("Could not find assets/icon.png");
+    let image = ico::IconImage::read_png(file).expect("Could not read PNG file");
+    icon_dir.add_entry(
+        ico::IconDirEntry::encode(&image).expect("Could not add PNG file to icon collection"),
+    );
+    // Write the ICO file to disk
+    let file = std::fs::File::create("assets/icon.ico").expect("Could not create assets/icon.ico");
+    icon_dir
+        .write(file)
+        .expect("Could not write assets/icon.ico to disk.");
+    // Compile and link icon.rc
+    embed_resource::compile("assets/icon.rc", embed_resource::NONE);
```

Comment:
Since it's doing some additional work now instead of just calling `embed_resource::compile(...)`, this should all be behind a check that the platform is Windows, because otherwise it's wasting time doing work that's useless.

I think (hope) this would be as simple as surrounding it with
```rust
if env::var_os("CARGO_CFG_WINDOWS").is_some() {
    ...
}
```

Reply:
Just surrounding the code like you suggested appears to do the job, it recognizes I am on Windows. I'll test it on a linux vm before pushing.

Reply:
Feel free to push without checking in a VM if it's gonna take any time, I run Linux and can test it myself if that'll be quicker/easier.

Reply:
Functions as intended on windows and ubuntu.

(Don't worry, it was fun getting ubuntu working for the first time)